tests/test_cli/test_add/test_contract_dependency.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2021 Valory AG
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This test module contains the tests for the `aea create` sub-command."""
+
+import os
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import List
+
+from click.testing import Result
+
+from aea.cli import cli
+from aea.configurations.base import AgentConfig
+from aea.configurations.constants import DEFAULT_AEA_CONFIG_FILE
+from aea.configurations.data_types import PackageType, PublicId
+from aea.configurations.loader import ConfigLoader
+from aea.test_tools.click_testing import CliRunner
+
+from tests.conftest import AUTHOR, CLI_LOG_OPTION, ROOT_DIR
+
+
+REGISTRY_PATH = Path(ROOT_DIR) / "tests" / "data" / "packages"
+
+
+class TestCreate:
+    """Test that the command 'aea create <agent_name>' works as expected."""
+
+    def _load_agent_config(self) -> AgentConfig:
+        """Load agent config for current dir."""
+        agent_loader = ConfigLoader.from_configuration_type(PackageType.AGENT)
+        with open(str(self.agent_dir / DEFAULT_AEA_CONFIG_FILE), "r") as fp:
+            agent_config = agent_loader.load(fp)
+        return agent_config
+
+    def _run_command(self, options: List) -> Result:
+        """Run command with default options."""
+        result = self.runner.invoke(
+            cli, [*CLI_LOG_OPTION, f"--registry-path={str(REGISTRY_PATH)}", *options],
+        )
+        assert result.exit_code == 0, result.stdout
+        return result
+
+    def test_run(self):
+        """Run the test."""
+
+        temp_dir = Path(TemporaryDirectory().name)
+        temp_dir.mkdir(exist_ok=True)
```

Comment:
The test was crashing on the windows because the process was not able to delete the temporary directory in the cleanup. So we're not deleting the temporary directory (I'm assuming the allocated resources are being deleted after the CI runs )